include/json/allocator.h

```diff
@@ -38,8 +45,12 @@ template <typename T> class SecureAllocator {
    * The memory block is filled with zeroes before being released.
    */
   void deallocate(pointer p, size_type n) {
-    // memset_s is used because memset may be optimized away by the compiler
-    memset_s(p, n * sizeof(T), 0, n * sizeof(T));
+    #if defined(_MSC_VER)
+      RtlSecireZeroMemory(p, n * sizeof(T));
+    #else
+      for (auto it = reinterpret_cast<volatile unsigned char* volatile>(p); n; ++it, --n)
+        *it = 0;
```

Comment:
It's just a nitpick, but why not:

```cpp
auto begin = reinterpret_cast<volatile unsigned char* volatile>(p);
std::copy(begin, begin + n, 0);
```

It's a little bit lighter on the eye while reading and will evaluate to nearly the same loop.

Reply:
std::fill is available since c++17 this won't work for everyone

Reply:
> std::fill is available since c++17 this won't work for everyone

not true. It's an old function

Reply:
didn't mean to over-engineer this single line, but `std::fill_n(begin, n, 0);` is pretty declarative and available even in c++11.

Reply:
Fixed it !